### PR TITLE
proposals from RPN per image during training

### PR DESCRIPTION
**Pull request by ChenJoya**
Hi @fmassa , the function ```select_over_all_levels``` is modified for single image during training.
The original issue: https://github.com/facebookresearch/maskrcnn-benchmark/issues/672
Thank you ^ ^

### CODE REVIEW

**Comment by ChenJoya** [comment]
Hi @fmassa , you are right, and the change of code needs to be check more.
Therefore I only rectify the README.md, and show how to set ```FPN_POST_NMS_TOP_N_TRAIN```. 
Thanks for your advise!

**Comment by rodrigoberriel** [comment]
@fmassa Would it be okay if we add a flag (e.g., `_C.MODEL.RPN.AS_IN_DETECTRON` -- hopefully a better name) with default `True` and add a section (e.g., "Known Issues" or "Known Bugs") to the README describing the default behavior, known side-effects (the one described in this PR) and the workarounds (1. do as recommended in this PR if the user cannot afford to re-train or 2. train in the correct way changing the flag)? Or maybe just a section to the `TROUBLESHOOTING.md`? I feel like we should provide a "switch" to avoid training like this, since training in the correct way apparently leads to the "same" results; then, it would be up to the user.

What do you think, @ChenJoya ?

File: maskrcnn_benchmark/modeling/rpn/inference.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def select_over_all_levels(self, boxlists):
         # and not per batch
         if self.training:
             objectness = torch.cat(
-                [boxlist.get_field("objectness") for boxlist in boxlists], dim=0
+               [boxlist.get_field("objectness") for boxlist in boxlists], dim=0
```
Comment (fmassa) [inline maskrcnn_benchmark/modeling/rpn/inference.py]: just removing this extra space change and this should be good to merge!

**Comment by fmassa** [comment]
@rodrigoberriel the documentation, after @ChenJoya improvements, already makes it fairly clear about this behavior.

Adding one more config argument would be an option to accommodate both behaviors, but it will just make the code more complicated.

**Comment by rodrigoberriel** [comment]
@fmassa fair enough about the documentation

Would you be willing to review a PR trying to accommodate both behaviors? I think it would be a 5~10 lines change and we could convert this TODO into a NOTE :)

https://github.com/facebookresearch/maskrcnn-benchmark/blob/1714b7c20d919f8c536a577e15ba3a03ce733cde/maskrcnn_benchmark/modeling/rpn/inference.py#L154-L159

**Comment by ChenJoya** [comment]
@fmassa  Thanks for your review. The extra space change has been removed :)

**Comment by fmassa** [comment]
@rodrigoberriel sure, why not adding an option (defaulting to the current behavior) that switches between behaviors. PRs welcome! :-)

**Pull request merged**